src/DependencyInjection/LazyContainerFactory.php

```diff
@@ -729,9 +729,12 @@ static function (
             }
 
             $skippedClassResolver = new SkippedClassResolver();
-            $skippedClasses = array_keys($skippedClassResolver->resolve());
-            foreach ($skippedClasses as $skippedClass) {
-                ContainerMemento::forgetService($container, $skippedClass);
+            $skippedElements = $skippedClassResolver->resolve();
+            foreach ($skippedElements as $skippedClass => $path) {
+                // completely forget the Rector rule only when no path specified
+                if ($path === null) {
+                    ContainerMemento::forgetService($container, $skippedClass);
+                }
             }
 
             $hasForgotten = true;
```

Comment:
is this flag still correct then?

Reply:
That seems used on line 727 to run once on phpunit